cms/djangoapps/contentstore/exams.py

```diff
@@ -0,0 +1,123 @@
+"""
+Code related to working with the exam service
+"""
+
+import json
+import logging
+
+import requests
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from edx_rest_api_client.auth import SuppliedJwtAuth
+
+from openedx.core.djangoapps.course_apps.toggles import exams_ida_enabled
+from openedx.core.djangoapps.oauth_dispatch.jwt import create_jwt_for_user
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from .views.helpers import is_item_in_course_tree
+
+log = logging.getLogger(__name__)
+User = get_user_model()
+
+
+def register_exams(course_key):
+    """
+    This is typically called on a course published signal. The course is examined for sequences
+    that are marked as timed exams. Then these are registered with the exams service.
+    Likewise, if formerly registered exams are not included in the payload they will
+    be marked inactive by the exam service.
+    """
+    if not settings.FEATURES.get('ENABLE_SPECIAL_EXAMS') or not exams_ida_enabled(course_key):
+        # if feature is not enabled then do a quick exit
+        return
+
+    course = modulestore().get_course(course_key)
+    if course is None:
+        raise ItemNotFoundError("Course {} does not exist", str(course_key))  # lint-amnesty, pylint: disable=raising-format-tuple
+
+    # get all sequences, since they can be marked as timed/proctored exams
+    _timed_exams = modulestore().get_items(
+        course_key,
+        qualifiers={
+            'category': 'sequential',
+        },
+        settings={
+            'is_time_limited': True,
+        }
+    )
+
+    # filter out any potential dangling sequences
+    timed_exams = [
+        timed_exam
+        for timed_exam in _timed_exams
+        if is_item_in_course_tree(timed_exam)
+    ]
+
+    exams_list = []
+    locations = []
+    for timed_exam in timed_exams:
+        location = str(timed_exam.location)
+        msg = (
+            'Found {location} as an exam in course structure.'.format(
+                location=location
+            )
+        )
+        log.info(msg)
+        locations.append(location)
+
+        if timed_exam.is_proctored_exam:
```

Comment:
[question/suggestion]: Could there be any potential issues in the future with using conditional statements like this and also with defining the exam types here? Is there an enum/class/other data type that could be better used here?

Reply:
oh good observation, my thinking is we would eventually move away from using these independent variables and just directly set exam type. They don't serve any purpose in the LMS/CMS other than backwards compatibility. Maybe I should move this into a function just in case it needs to be referenced elsewhere.